buildSrc/src/main/groovy/corda.java-only.gradle

```
@@ -1,9 +1,3 @@
-pluginManager.withPlugin('org.jetbrains.kotlin.jvm') {
+if (project.sourceSets.main.output.files.toString().contains("\\classes\\kotlin\\")) {
     throw new StopExecutionException("Module '${project.path}' should only contain Java classes")
-}
-
-pluginManager.withPlugin('java') {
-    java {
-        withJavadocJar()
-    }
-}
+}
```

Maybe rewrite more along lines of:
```gradle
plugins {
    id 'org.jetbrains.kotlin.jvm'
}

tasks.matching { it.name ==~ 'compile.*Kotlin' }.configureEach {
    doFirst {
        throw new InvalidUserCodeException("Module '${project.path}' should only contain Java classes")
    }
}
```
The "root" `build.gradle` file should already be applying the `dokka` plugin to anything that applies `org.jetbrains.kotlin.jvm`, so it's just a matter of preventing anyone from compiling Kotlin classes.

This is simpler than I thought - the Kotlin compile tasks don't execute if Gradle doesn't find any source files, and we don't want this to fail silently.

---

Ah nice, updated thanks